apple/macbook-pro/12-1/default.nix

```diff
@@ -7,6 +7,20 @@
     <nixpkgs/nixos/modules/hardware/network/broadcom-43xx.nix>
   ];
 
+  powerManagement = {
+    cpuFreqGovernor = lib.mkDefault "conservative";
+
+    powerUpCommands = ''
+      ${pkgs.kmod}/bin/modprobe brcmfmac
+      ${pkgs.coreutils}/bin/sleep 5
+      ${pkgs.systemd}/bin/systemctl restart wpa_supplicant.service
```

Comment:
We need something better here: There are two different wpa supplicant modules that have different service names. Than there is network manager with its own wpa_supplicant and iwd, which independent from network manager. What does actually happen if the wifi interface disappears/appears? Does wpa supplicant not pick it up again?

Reply:
found what happens: when unloading a module, wpa_supplicant simply stops.

```cosole
Nov 27 13:57:31 mac sudo[10347]: sevenfourk : TTY=pts/2 ; PWD=/home/sevenfourk/Sources/GitHub/nixos-hardware ; USER=root ; COMMAND=/run/current-system/sw/bin/m
odprobe -r brcmfmac
Nov 27 13:57:31 mac sudo[10347]: pam_unix(sudo:session): session opened for user root by (uid=0)
Nov 27 13:57:31 mac kernel: usbcore: deregistering interface driver brcmfmac
Nov 27 13:57:31 mac kernel: ieee80211 phy1: brcmf_fil_cmd_data: bus is down. we have nothing to do.
Nov 27 13:57:31 mac kernel: ieee80211 phy1: brcmf_cfg80211_get_tx_power: error (-5)
Nov 27 13:57:31 mac dhcpcd[1524]: wlp3s0: fe80::52c7:bfff:fe0d:2c92 is unreachable
Nov 27 13:57:31 mac wpa_supplicant[10086]: wlp3s0: CTRL-EVENT-DISCONNECTED bssid=50:c7:bf:0d:2c:91 reason=1 locally_generated=1
Nov 27 13:57:31 mac wpa_supplicant[10086]: dbus: wpa_dbus_property_changed: no property SessionLength in object /fi/w1/wpa_supplicant1/Interfaces/0
Nov 27 13:57:31 mac dhcpcd[1524]: arp_read: wlp3s0: Network is down
Nov 27 13:57:31 mac dhcpcd[1524]: wlp3s0: carrier lost
Nov 27 13:57:31 mac systemd[1]: Stopping WPA Supplicant...
```

i always used the simplest method: only wpa_supplicant, do you think i should better use network manager or so?

Reply:
I use iwd because it's interface is nicer than wpa_supplicant. It works both with and without network manager. Maybe we should mention in the README that one needs to restart wpa_supplicant somehow if this is the case?

Reply:
yes, perhaps documenting it would be the best idea, we can't restart the service, someone might not even use it.

Reply:
added a README.md